packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -120,6 +120,63 @@ module.exports = {
     }
   },
 
+  async changePassword(ctx) {
+    const params = _.assign({}, ctx.request.body, ctx.params);
+
+    if (!_.get(ctx, 'state.user.id')) {
+      throw new ApplicationError('You must be authenticated to reset your password');
+    }
+
+    if (
+      params.currentPassword &&
+      params.password &&
+      params.passwordConfirmation &&
+      params.passwordConfirmation === params.password
+    ) {
+      const user = await strapi.entityService.findOne(
+        'plugin::users-permissions.user',
+        ctx.state.user.id
+      );
+
+      if (!user) {
+        throw new ValidationError('User not found');
+      }
+
+      const validPassword = await getService('user').validatePassword(
```

Comment:
if you setup the validations I think you can remove part of this. Have a look at the register I think It should be very similar